### ISSUE TITLE
Bind {namespace} to NamespaceName

### ISSUE THREAD

**Issue by wslulciuc** [MarquezProject/marquez#374]
Let's update all [resources](https://github.com/MarquezProject/marquez/tree/master/src/main/java/marquez/api/resources) to bind the path variable `{namespace}` to the type [`NamespaceName`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/common/models/NamespaceName.java)

**Comment by pbrahmbhatt3** [comment]
can i work on it?

**Comment by wslulciuc** [comment]
Sure! @pbrahmbhatt3 👍

**Comment by pbrahmbhatt3** [comment]
@wslulciuc Hi, I was working on this bug and I updated DatasetResource.java file. However, gradlew test failed with error:

![dataresourcetest java file](https://user-images.githubusercontent.com/27025266/53911013-4bb99f80-4023-11e9-8d7f-03ee7da28b3e.png)

I figured out that the datasetResourceTest.java file was passing a string argument for datasetResource.List instead of passing a NamespaceName obj. Here is the list object from DatasetResource.Java file

![dataresource](https://user-images.githubusercontent.com/27025266/53911182-b7037180-4023-11e9-86bb-23e64b800c5d.png)

Hence I updated the datasetResourceTest.java file to:
![dataresourcetest java file](https://user-images.githubusercontent.com/27025266/53911299-08abfc00-4024-11e9-8a50-41bb73ac4eee.png)

Now it works. Your advice on this will be much appreciated. Thank you. :)
